### ISSUE TITLE
feat(semantic): error on invalid assignment target

### ISSUE THREAD

**Issue by Boshen** [Boshen/oxc#326]
The current AST need to explicitly check for invalid non-IdentifierReference and non-MemberExpression in the four TSXXX expressions from `SimpleAssignmentTarget` below. 

```rust
#[derive(Debug, PartialEq, Hash)]
#[cfg_attr(feature = "serde", derive(Serialize), serde(untagged))]
pub enum SimpleAssignmentTarget<'a> {
    AssignmentTargetIdentifier(Box<'a, IdentifierReference>),
    MemberAssignmentTarget(Box<'a, `MemberExpression`<'a>>),
    TSAsExpression(Box<'a, TSAsExpression<'a>>),
    TSSatisfiesExpression(Box<'a, TSSatisfiesExpression<'a>>),
    TSNonNullExpression(Box<'a, TSNonNullExpression<'a>>),
    TSTypeAssertion(Box<'a, TSTypeAssertion<'a>>),
}
```

e.g. `(x+1 as z) = 1` should raise the error `The left-hand side of an assignment expression must be a variable or a property access.(2364)` because the assignment target is not `IdentifierReference` nor `MemberExpression`.

We need to add a new `crates/oxc_semantic/src/checker/typescript.rs` for checking TypeScript syntax.

**Comment by mysteryven** [comment]
I am interested in it and will try it these days.
